### PR TITLE
docs: added 4.1 upgrade doc

### PR DESCRIPTION
**Pull request by kaustubhkapatral**
## Description

Closes: #XXXX

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/cosmos/cosmos-sdk/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] followed the guidelines for [building modules](https://github.com/cosmos/cosmos-sdk/blob/main/docs/building-modules)
- [ ] included the necessary unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/main/CONTRIBUTING.md#testing)
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed 
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

File: docs/validators/upgrades/v4.1-upgrade.md [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+```bash
+v4.1.2
+```
+
+Create a `v4.1.0` directory and copy the upgrade binary (`v4.1.2`) to the directory:
+
+```bash
+mkdir -p $HOME/.regen/cosmovisor/upgrades/v4.1.0/bin
+cp ./build/regen $HOME/.regen/cosmovisor/upgrades/v4.1.0/bin
```
Comment (anilCSE) [inline docs/validators/upgrades/v4.1-upgrade.md:62]:
```suggestion
cp ./build/regen $HOME/.regen/cosmovisor/upgrades/v4.1.0/bin
$HOME/.regen/cosmovisor/upgrades/v4.1.0/bin/regen version # should be v4.1.2
```

**Review by anilCSE — CHANGES_REQUESTED** [review]

**Comment by aleem1314** [comment]
Let's add an entry for v4.1 in upgrade [overview](https://github.com/regen-network/regen-ledger/blob/main/docs/validators/upgrades/README.md?plain=1#L7).

**Review by anilCSE — APPROVED** [review]
lgtm

**Review by aleem1314 — APPROVED** [review]
lgtm

**Comment by ryanchristo** [comment]
@mergifyio backport release/v4.1.x

**Pull request merged**